code/game/objects/structures/flora.dm

```diff
@@ -1,48 +1,219 @@
-//trees
+/obj/structure/flora
+	anchored = TRUE
+	var/health = 25
+	var/health_max = 25
+	var/on_fire = FALSE
+
+/obj/structure/flora/ex_act(severity)
+	switch(severity)
+		if(1)
+			cdel(src)
+		if(2)
+			if(prob(70))
+				cdel(src)
+		if(3)
+			if(prob(50))
+				cdel(src)
+
+/obj/structure/flora/Dispose()
+	processing_objects.Remove(src)
+	return ..()
+
+/obj/structure/flora/attackby(obj/item/W, mob/living/user)
+	if(!W || !user || isnull(W) || (W.flags_item & NOBLUDGEON))
+		return 0
+
+	var/damage = W.force
+	if(W.w_class < 4 || !W.sharp || W.force < 20) //only big strong sharp weapon are adequate
+		damage *= 0.25
+
+	if(istype(W, /obj/item/tool/weldingtool))
+		var/obj/item/tool/weldingtool/WT = W
+
+		if(WT.remove_fuel(0))
```

Comment:
No fuel cost? Eye protection not needed?

Reply:
i missed the fuelcost part, that'll be changed, about the eye damage, does attacking a mob with a welder hurt eyes?